packages/features/users/components/UserTable/UserListTable.tsx

```diff
@@ -0,0 +1,353 @@
+import { trpc } from "@calcom/trpc";
+import { Avatar, Badge, Button, ConfirmationDialogContent, DataTable, Dialog, showToast } from "@calcom/ui";
+
+import { ChangeUserRoleModal } from "./ChangeUserRoleModal";
+import { ImpersonationMemberModal } from "./ImpersonationMemberModal";
+import { InviteMemberModal } from "./InviteMemberModal";
+import { TableActions } from "./UserTableActions";
+
+export interface User {
+  id: number;
+  username: string | null;
+  email: string;
+  timeZone: string;
+  role: MembershipRole;
+  accepted: boolean;
+  disableImpersonation: boolean;
+  teams: {
+    id: number;
+    name: string;
+    slug: string | null;
+  }[];
+}
+
+type Payload = {
+  showModal: boolean;
+  user?: User;
+};
+
+export type State = {
+  changeMemberRole: Payload;
+  deleteMember: Payload;
+  impersonateMember: Payload;
+  inviteMember: Payload;
+};
+
+export type Action =
+  | {
+      type: "SET_CHANGE_MEMBER_ROLE_ID" | "SET_DELETE_ID" | "SET_IMPERSONATE_ID" | "INVITE_MEMBER";
+      payload: Payload;
+    }
+  | {
+      type: "CLOSE_MODAL";
+    };
+
+const initialState: State = {
+  changeMemberRole: {
+    showModal: false,
+  },
+  deleteMember: {
+    showModal: false,
+  },
+  impersonateMember: {
+    showModal: false,
+  },
+  inviteMember: {
+    showModal: false,
+  },
+};
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "SET_CHANGE_MEMBER_ROLE_ID":
+      return { ...state, changeMemberRole: action.payload };
+    case "SET_DELETE_ID":
+      return { ...state, deleteMember: action.payload };
+    case "SET_IMPERSONATE_ID":
+      return { ...state, impersonateMember: action.payload };
+    case "INVITE_MEMBER":
+      return { ...state, inviteMember: action.payload };
+    case "CLOSE_MODAL":
+      return {
+        ...state,
+        changeMemberRole: { showModal: false },
+        deleteMember: { showModal: false },
+        impersonateMember: { showModal: false },
+        inviteMember: { showModal: false },
+      };
+    default:
+      return state;
+  }
+}
+
+export function UserListTable() {
+  const { data: session } = useSession();
+  const { data: currentMembership } = trpc.viewer.organizations.listCurrent.useQuery();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { t } = useLocale();
+  const utils = trpc.useContext();
+
+  const { data, isLoading, fetchNextPage, isFetching } =
+    trpc.viewer.organizations.listMembers.useInfiniteQuery(
+      {
+        limit: 10,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        keepPreviousData: true,
+      }
+    );
+
+  const adminOrOwner = currentMembership?.user.role === "ADMIN" || currentMembership?.user.role === "OWNER";
+
+  const memorisedColumns = useMemo(() => {
+    const permissions = {
+      canEdit: adminOrOwner,
+      canRemove: adminOrOwner,
+      canImpersonate: false,
+    };
```

Comment:
Set of base permissions that we expand upon when rendering each row for table actions